src/configuration.cpp

```diff
@@ -356,6 +355,62 @@ string Configuration::convert_svg(string icon_filename)
   return converted;
 }
 
+/*
+ *  localize_icon()
+ *
+ *  Try to find a localized version of the icon filename. Otherwise return the default one.
+ *  It works by querying the LANG environment variable, so it's easy to test.
+ *
+ *  Example, given LANG is set to "es_AR.UTF-8":
+ *
+ *     /usr/share/kano-desktop/icons/myicon.png ... would become:
+ *     /usr/share/kano-desktop/icons/i18n/es_AR/myicon.png
+ *
+ */
+string Configuration::localize_icon(string icon_filename)
+{
+    // Obtain current locale, return original file if unavailable
+    char *current_locale = getenv("LANG");
+    if (!current_locale || !strlen(current_locale))
+        return icon_filename;
+    else {
+        char *dot = strchr(current_locale, '.');
+        if (dot)
+            *dot = 0x00;
+    }
+
+    // Split the path and filename (basename / basedir are harmful)
+    char localized[PATH_MAX], filename[PATH_MAX], path[PATH_MAX];
+    strcpy (localized, icon_filename.c_str());
+    char *path_ext=strrchr(localized, '/');
```

Comment:
`std::string` has a [`find_last_of`](http://www.cplusplus.com/reference/string/string/find_last_of/) method which will both simplify this and make it a little safer when combined with [`substr`](http://www.cplusplus.com/reference/string/string/substr/).